src/django_upgrade/data.py

```diff
@@ -21,6 +21,8 @@ def __init__(self, target_version: tuple[int, int]) -> None:
 
 test_re = re.compile(r"(\b|_)tests?(\b|_)")
 
+dunder_init_re = re.compile(r"\b__init__.py$")
```

Comment:
I think we should check for a path separator rather a word boundary (which would match weird names like `foo-__init__.py`)

something like:

```suggestion
dunder_init_re = re.compile(r"(^|[\/])__init__.py$")
```

Reply:
Oh yes, of course... I think I didn't completely undertsand the regex 🙈 

I can make the changes, but I see you started to pick them up?